ReactiveCocoa/DelegateProxy.swift

```diff
@@ -0,0 +1,36 @@
+import ReactiveSwift
+import enum Result.NoError
+
+internal class DelegateProxy<Delegate: NSObjectProtocol>: NSObject {
+	internal weak var forwardee: Delegate? {
+		didSet {
+			originalSetter(self)
+		}
+	}
+
+	internal var interceptedSelectors: Set<Selector> = []
+
+	private let originalSetter: (AnyObject) -> Void
+
+	init(_ originalSetter: @escaping (AnyObject) -> Void) {
+		self.originalSetter = originalSetter
+	}
+
+	override func forwardingTarget(for selector: Selector!) -> Any? {
+		return interceptedSelectors.contains(selector) ? nil : forwardee
+	}
+
+	func intercept(_ selector: Selector) -> Signal<(), NoError> {
+		interceptedSelectors.insert(selector)
+		originalSetter(self)
```

Comment:
A lot of Cocoa classes cache information about delegates. Since we only intercept a subset but not all, and the user supplied delegate might be changed, we need to force the delegater to update their cache by invoking the setter again.